elytron/src/main/java/org/wildfly/extension/elytron/DynamicSSLContextHelper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.extension.elytron;
+
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.dynamic.ssl.DynamicSSLContext;
+import org.wildfly.security.dynamic.ssl.DynamicSSLContextImpl;
+import org.wildfly.security.dynamic.ssl.DynamicSSLContextException;
+
+import javax.net.ssl.SSLContext;
+import java.security.GeneralSecurityException;
+import static org.wildfly.extension.elytron._private.ElytronSubsystemMessages.ROOT_LOGGER;
+
+/**
+ * Helper class for obtaining an instance of DynamicSSLContext created from the provided AuthenticationContext
+ */
+class DynamicSSLContextHelper {
+
+    /**
+     * Get DynamicSSLContext instance from the provided authentication context
+     * @param authenticationContext authentication context to use with the DynamicSSLContext
+     * @return DynamicSSLContext instance
+     */
+    static SSLContext getDynamicSSLContextInstance(AuthenticationContext authenticationContext) {
+        try {
+            return new DynamicSSLContext(new DynamicSSLContextImpl(authenticationContext));
```

Comment:
How about if they provision at the Default stability level but then try and run at community level - do we need to handle an error for when the module does not exist?

Reply:
@darranl Hm maybe I can use Class.forName instead of imports, and catch the ClassNotFoundException if that sounds good?

Reply:
Would `getDynamicSSLContextInstance` get invoked if the dynamic SSL resource management resource is not registered?

Reply:
My understanding is this method won't be called when you are running on `default` since this is supplied by the add handler that provides the Dynamic SSL context, but that handler will never be used because the resource won't be available on a lower stability level.

Reply:
@yersan That was my original understanding as well. That the class DynamicSSLContextHelper would not be loaded if dynamic SSL resource is not registered... but was not completely sure

Reply:
FYI there is a thread in Zulip with some more context but the general scenario is, what happens when a user provisions a server at the default stability level but then runs at the community level and tries to use this resource:

https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/Disable.20Specific.20Stability.20Levels.20at.20Provisioning.20Time/near/432858502